metabase/metabase
# Fix milestone reminder

### Description

Resolves #42835.

The milestone reminder workflow has been broken.
My previous attempt to fix it didn't really work because the parsing logic was wrong.

This is the example:
https://github.com/metabase/metabase/actions/runs/9382661958/job/25834555041#step:2:78

Notice how it tries to fetch `https://api.github.com/repos/metabase/metabase/issues/s`

Some local experimentation explains why this happened in the past:
![image](https://github.com/metabase/metabase/assets/31325167/ccccc597-7a03-4c72-9a76-ba7010d78204)

I've now included a lot of unit tests to make sure this script is working.

### Notable Changes
- The script is now able to parse multiple linked issues, which wasn't the case before
- The issue number with be included in the bot message ([check this comment](https://github.com/metabase/metabase/pull/43684#issuecomment-2150902966)) 

### How to verify
- Run accompanied unit tests
- Check this test PR out: https://github.com/metabase/metabase/pull/43684
    - It links to one issue with and one without the milestone, and the bot correctly parses the issue without the milestone

### Checklist
